src/app/modules/payments/payments.service.ts

```diff
@@ -330,3 +330,24 @@ export const findLatestSuccessfulPaymentByEmailAndFormId = (
     return okAsync(result)
   })
 }
+
+/**
+ * Retrieves all payments that are in Pending or Failed statuses.
+ * @returns a list of payments that are incomplete.
+ */
+export const getIncompletePayments = (): ResultAsync<
+  IPaymentSchema[],
+  DatabaseError
+> => {
+  return ResultAsync.fromPromise(
+    PaymentModel.getByStatus(PaymentStatus.Pending, PaymentStatus.Failed),
```

Comment:
What's the rationale for getting failed payments again? (apologies, this is probably a RTFM on me 😰 )

Reply:
failed payments can be retried!